modules/createContext.js

```diff
@@ -0,0 +1,162 @@
+import React from "react";
+import PropTypes from "prop-types";
+import warning from "warning";
+
+const valueTypes = {
+  string: PropTypes.string,
+  number: PropTypes.number,
+  function: PropTypes.func,
+  boolean: PropTypes.bool
+};
+
+// TODO: This could probably be improved.
+function getPropType(value) {
+  const type = typeof value;
+
+  if (type === "object") {
+    return Array.isArray(value) ? PropTypes.array : PropTypes.object;
+  }
+
+  return valueTypes[type] || PropTypes.any;
```

Comment:
This should probably **always** be any. You can’t infer the type from the value. E.g. the default value could be `42` and the type could still be `string | number`

Reply:
Ah, good point. That code was just guessing anyway. Happy to get rid of it.